src/marqo/s2_inference/s2_inference.py

```diff
@@ -115,19 +127,41 @@ def _create_model_cache_key(model_name: str, device: str, model_properties: dict
 def _update_available_models(model_cache_key: str, model_name: str, validated_model_properties: dict,
                              device: str,
                              normalize_embeddings: bool) -> None:
-    """loads the model if it is not already loaded
+    """loads the model if it is not already loaded.
+    Note this method assume the model_properties are validated.
     """
+    model_size = get_model_size(model_name, validated_model_properties)
```

Comment:
this doesn't need to be calculated if the model is already loaded right? maybe add under `if model_cache_key not in available_models:`